### PR TITLE
Make coveralls work with coverage 4 alpha.

### PR DESCRIPTION
**Pull request by jessamynsmith**
Here is a first cut at making coveralls handle the alpha version of coverage 4. This was enough for me to get template coverage working on coveralls.io: https://coveralls.io/builds/1934440

This is related to issue #47

### CODE REVIEW

**Comment by coagulant** [comment]
Awesome, I didn't realize template coverage would work with coveralls.
I'm definately interested in coverage 4a support, just need some time for review

**Comment by jessamynsmith** [comment]
Very understandable -- it's not a trivial change. I'm certainly open to modifying it. It would also be great if @nedbat took a look. :)

One thought: If possible it would be great to have tox test against coverage 3.x and 4.x.

**Comment by jessamynsmith** [comment]
I just added another unit test for malformed files on newer python and/or coverage.

File: coveralls/reporter.py [hunk 1]
```diff
@@ -55,20 +56,23 @@ def get_hits(self, line_num, analysis):
...
-                encoding = source_encoding(source)
-                if encoding != 'utf-8':
-                    source = source.decode(encoding).encode('utf-8')
-        except UnicodeDecodeError:
-            log.warn('Source file %s can not be properly decoded, skipping. '
-                     'Please check if encoding declaration is ok', basename(cu.filename))
-            return
-        finally:
-            source_file.close()
+        if __version__.startswith('3.'):
```
Comment (coagulant) [inline coveralls/reporter.py]: I'd rather try detecting objects and not version sniffing.
I suppose we can inspect analysis.file_reporter and see if it exists

File: tox.ini [hunk 2]
```diff
@@ -7,5 +7,6 @@ deps =
     mock
     sh
     pyyaml: PyYAML>=3.10
+    cov4: coverage>=4.0a4
```
Comment (coagulant) [inline tox.ini]: I guess we need to be explicit and force coverage 3.x for another env to be future compatible when coverage 4.x will become default.

Maybe like that
```
{py26,py27,py32,py33,py34,pypy,pypy3}-{cov3,cov4 {py26,py27,py32,py33,py34,pypy,pypy3}-pyyaml
```

File: coveralls/reporter.py (hunk 1 above)
Comment (jessamynsmith) [inline coveralls/reporter.py]: Agreed. I had actually written it that way at first, I only changed it to match the version sniffing for python. I've updated the diff.

File: tox.ini (hunk 2 above)
Comment (jessamynsmith) [inline tox.ini]: Updated as suggested.

**Comment by coagulant** [comment]
Thanks a lot. I'm planning to make 1.0a1 release sometime this week.

**Comment by jessamynsmith** [comment]
Great! :)

**Comment by nedbat** [comment]
I am really glad to see this work happening, though I wish I had time to keep up with the changes.  I still have big changes to make to coverage.py, so an alpha release is a good idea.

**Pull request merged**